### PR TITLE
Add page to create a new programming environment

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Quick and dirty page to add a new programming environment! I also added some name validations for ProgrammingEnvironment as part of this PR.

https://user-images.githubusercontent.com/46464143/153047957-734de25b-778d-42f1-9004-100ff5e9fc52.mov

## Testing story

ruby unit tests. I couldn't think of a good test for the simple form but happy to add one if folks think it would be useful 

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/levelbuilder/code-docs-editor/NewProgrammingEnvironmentForm.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        builder to create code documentation.
+      </h2>
+      <label>
+        Programming Environment Slug
+        <input name="name" style={styles.inputStyle} required />
+        <HelpTip>
+          <p>
+            The programming environment slug is used in URLs and cannot be
+            updated once set. A slug can only contain lowercase letters,
+            numbers, underscores, and dashes. Likely, this should be the same at
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/code-docs-editor/NewProgrammingEnvironmentForm.jsx]: Should we allow both underscores and dashes or just pick one?

**Review by dmcavoy — APPROVED** [review]
Thanks for adding this!

File: apps/src/lib/levelbuilder/code-docs-editor/NewProgrammingEnvironmentForm.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/code-docs-editor/NewProgrammingEnvironmentForm.jsx]: I was debating that and feeling a bit lazy so I left both. But now I'm thinking we should pick dashes and remove underscores. If we need them later, it'll be easy to allow them.

Comment (dmcavoy) [reply apps/src/lib/levelbuilder/code-docs-editor/NewProgrammingEnvironmentForm.jsx]: That sounds good me

**Pull request merged**